drivers/esp8266/sock/udp.c

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (C) 2016 Hamburg University of Applied Sciences
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     driver
+ * @{
+ *
+ * @file
+ * @brief       Implementation of the UDP socket with the esp8266 802.11b/g/n radio module
+ *              In the implementation we assume that we are only woking with exactly one ESP8266 module.
+ *              The implementation follows the monitor pattern, thereby allow only one udp action at a time.
+ *
+ * @author      Markus Blechschmidt <markus.blechschmidt@haw-hamburg.de>
+ *
+ * @}
+ */
+
+#include "net/sock/udp.h"
+#include "esp8266.h"
+#include "xtimer.h"
+#include "thread.h"
+#include "mutex.h"
+#include "msg.h"
+#include "net/ipv4/addr.h"
+
+#define ENABLE_DEBUG  (1)
+#include "debug.h"
+
+/* private varibaled fo the driver */
+static ESP8266_t esp8266;
+static char threadstack[THREAD_STACKSIZE_DEFAULT];
```

Comment:
when allowing debug, this needs to be large. You can conveniently just add DEBUG_EXTRA_STACKSIZE, which will be (0) in case debugging is disabled.